SwiftUIBasics/Views/SignUpView.swift

```diff
@@ -52,6 +61,49 @@ class SignUpViewModel: ObservableObject {
             }
             .assign(to: \.isValidPasswordUpperCase, on: self)
             .store(in: &cancelableSet)
+        
+        $password
+            .receive(on: RunLoop.main)
+            .map {
+                password in
+                let pattern = "[0-9]"
+                if let _ = password.range(of: pattern, options: .regularExpression) {
+                    return true
+                } else {
+                    return false
+                }
+            }
+            .assign(to: \.isValidPasswordNumberCase, on: self)
+            .store(in: &cancelableSet)
+        
+        $password
+            .receive(on: RunLoop.main)
+            .map {
+                password in
+                let pattern = "[a-z]"
+                if let _ = password.range(of: pattern, options: .regularExpression) {
+                    return true
+                } else {
+                    return false
+                }
+            }
+            .assign(to: \.isValidPasswordLowerCase, on: self)
+            .store(in: &cancelableSet)
+        
+        $password
+            .receive(on: RunLoop.main)
+            .map {
+                password in
+                let pattern = "[^\\sA-Za-z0-9]+"
```

Comment:
this could be improved by
```suggestion
                let pattern = "[\\W]"
```